Add/update Ledger end-to-end guide (#82)
oasisprotocol/docs: docs/general/manage-tokens/holding-rose-tokens/ledger-wallet.md

# Self-Custody With Ledger Hardware Wallet

> How to use Ledger hardware wallet with Oasis Network (Oasis app) and Ledger Live software. 

Reviewer: This is a general page on using Ledger with Oasis. Ledger Live doesn't support Oasis tokens natively yet.

Reviewer: Perhaps, reword to add intro that this guide will help users set up their Ledger hardware wallet to use it with the Oasis Network.

> This guide is using Ledger Nano S (Secure Element 2.1.0, Microcontroller 1.12, Bootloaded 0.11), Ledger Live (version 2.40.2) and official [Oasis Wallet - Web](https://wallet.oasisprotocol.org).

Reviewer: This is a general page on how to set up Ledger for Oasis. I don't think this text is appropriate.

## Setup your Ledger device and Install Oasis App

> 1. To use your [Ledger](https://www.ledger.com) wallet to hold your ROSE tokens, you will have to install [Oasis app](https://github.com/Zondax/ledger-oasis) on your Ledger wallet via [Ledger Live](https://www.ledger.com/ledger-live/)'s Manager. You need to connect your Ledger to your device and unlock it with your PIN code first.

Reviewer: Put links at the end of sections and use something like:

```
Text .... [Ledger] ...
...
End of text ...

[Ledger]: https://www.ledger.com
```

> ![](<../../images/wallet/ledger/1_unlock_ledger.png>)

Reviewer 2: Fill in the image description (html's `alt` attribute).
```suggestion
![Unlock ledger](<../../images/wallet/ledger/1_unlock_ledger.png>)
```

> 2. Currently, you will have enable _Developer mode_ by going to _Settings -> Experimental features_ -> enable _Developer mode_ toggle, to be able to find the Oasis app in Ledger Live's Manager.

Reviewer 3: Strange inherited grammar: "you will have enable"

Author: "you will have TO enable" applied.

Author: "you will have TO enable" applied.

> ![](<../../images/wallet/ledger/2_ledger_live_experimental_developer_mode.png>)

Reviewer 2: Same here

> :::caution

Reviewer: Update versions to match info at https://docs.oasis.dev/oasis-core-ledger/usage/setup#installing-oasis-app-on-your-ledger-wallet

# Self-Custody With Ledger Hardware Wallet

This is a general documentation that will help users setup [Ledger] hadware wallets with Oasis Network. Ledger Live software doesn't support Oasis (ROSE) tokens natively yet. In this guide we will install Oasis app via Ledger Live to open and access wallet with one or multiple accounts via our official [Oasis Wallet - Web].

## Setup your Ledger device and Install Oasis App

1. To use your [Ledger] wallet to hold your ROSE tokens, you will have to install [Oasis app] on your Ledger wallet via [Ledger Live]'s Manager. You need to connect your Ledger to your device and unlock it with your PIN code first.

![Unlock ledger](<../../images/wallet/ledger/1_unlock_ledger.png>)

2. Currently, you will have to enable _Developer mode_ by going to _Settings -> Experimental features_ -> enable _Developer mode_ toggle, to be able to find the Oasis app in Ledger Live's Manager.

![Ledger Live -> Experimental features -> Developer Mode](<../../images/wallet/ledger/2_ledger_live_experimental_developer_mode.png>)

:::caution

The Oasis app requires an up-to-date firmware on your Ledger wallet:

> * At least [version 2.1.0](https://support.ledger.com/hc/en-us/articles/360010446000-Ledger-Nano-S-firmware-release-notes) released on Nov 30, 2021 on a Nano S device.

Reviewer: Please, move these version links at the end of the section as well.

Author: OK, now all links will be at the end of the section/doc.

Experimental features_ -> enable _Developer mode_ toggle, to be able to find
the Oasis app in Ledger Live's Manager.

![Ledger Live -> Experimental features -> Developer Mode](<../../images/wallet/ledger/2_ledger_live_experimental_developer_mode.png>)

:::caution

The Oasis app requires an up-to-date firmware on your Ledger wallet:

* At least [version 2.1.0] released on Nov 30, 2021 on a Nano S device.
* At least [version 2.0.0] released on Oct 21, 2021 on a Nano X device.

Follow Ledger's instructions for updating the firmware on your Ledger wallet:

* [Nano S]
* [Nano X]

:::

3. To find the Oasis app in the Ledger Live App catalog, you need to Allow
Ledger Manager on your Ledger device first, then you will be able to click App
catalog and search for _Oasis_:

![Allow Ledger Manager](<../../images/wallet/ledger/3_allow_ledger_manager.png>)
![Search app in catalog..](<../../images/wallet/ledger/4_search_apps.png>)

4. Install _Oasis_ app

![Install Oasis app](<../../images/wallet/ledger/6_search_results_oasis_install.png>)

:::info

You may see both the _Oasis_ app and the _OasisVal_ app come up in the search
results. You need to use the standard _Oasis_ app.

:::

> 5. After installation is completed, take your Ledger device, navigate to _Oasis_

Reviewer 2: ```suggestion
5. After the installation is completed, take your Ledger device, navigate to _Oasis_
```

app and use both buttons to open the app. A message _"Pending Ledger review"_

> will appear which you confirm with both buttons. Your Ledger device is ready

Reviewer 2: ```suggestion
will appear which you confirm by holding both buttons. Your Ledger device is ready
```

when you will see _"Oasis Ready"_ message.

![Oasis Ready](<../../images/wallet/ledger/6b_ledger_oasis_ready.png>)

:::caution

Make sure you install at least version 2.3.2 of the Oasis app.

:::

The Oasis app will use the [BIP 39] mnemonic seed stored on your Ledger wallet
to generate the private & public key pairs for your accounts.

## Manage Your Tokens

### Using Oasis Wallets

This is a simpler option since it allows you to connect to your Ledger wallet
via a web application or a browser extension.
	
In the example below we will use Oasis Wallet - Web. To learn more about other
features of the Oasis Wallets, please read the [Oasis Wallets] doc.

1. Navigate to [wallet.oasisprotocol.org] and click on the _Open wallet_ button.
Then, click on _Ledger_ when asked how to open your wallet.

![Oasis Wallet - Web -> Open wallet -> Ledger](<../../images/wallet/ledger/7_oasis_wallet_web_open_ledger.png>)

> 2. In the next step, click on the _Select the wallets to open_ button. Your

Reviewer 2: ```suggestion
2. In the next step, click on the _Select the accounts to open_ button. Your
```

browser will open a pop-up window where you will have to select your Ledger
device. Finally, click on _Connect_.

![Oasis Wallet - Web -> Select Ledger device and Connect](<../../images/wallet/ledger/8b_oasis_wallet_web_select_ledger_device_connect.png>)

:::caution

This step requires that your Ledger shows the _Oasis Ready_ message. After a
while your device may lock for safety reasons and you will need to unlock it to
perform this and subsequent steps.

:::

3. After connecting your Ledger to Oasis Wallet - Web another pop-up will appear
where you can choose to open one or more `oasis1` accounts derived from the seed
stored on your Ledger.

![Oasis Wallet - Web -> Select one or more accounts](<../../images/wallet/ledger/9_oasis_wallet_web_select_wallets_to_open.png>)

> 4. Your oasis1 wallet from Ledger device is opened and you can switch between

Reviewer 2: ```suggestion
4. The account from your Ledger device is now opened. If you import multiple accounts, you can switch between
```

> multiple wallets by clicking on your wallet address in the top-right corner.

Reviewer 2: ```suggestion
them by clicking on the account address in the top-right corner.
```

> Our demo wallet is empty.

Reviewer 2: ```suggestion
Our demo account is empty.
```

![Oasis Wallet - Web -> Selected accounts are opened](<../../images/wallet/ledger/10_oasis_wallet_web_opene_ledger_account.png>)

> 5. Now you can use your Ledger wallet to receive, send or delegate your ROSE.

Reviewer 2: ```suggestion
5. Now you can use your Ledger to receive, send or delegate ROSE.
```

> In our example, we have received 111 ROSE to Ledger wallet.

Reviewer 2: ```suggestion
In this example, we have received 111 ROSE to our Ledger account.
```

![Oasis Wallet - Web -> Exploring transactions](<../../images/wallet/ledger/11_oasis_wallet_web_received_rose_on_ledger_account.png>)

### Using Oasis CLI Tools

> This is the most powerful option that allows performing any token-related

Reviewer 2: ```suggestion
This is the most powerful option that allows performing not just token-related
```

> management task.

Reviewer 2: ```suggestion
tasks (sending, staking, ParaTime deposits and withdrawals), but also generating and/or signing
raw transactions, multi-signatures, network governance operations and so on.
```

# Self-Custody With Ledger Hardware Wallet

This is a general documentation that will help users setup [Ledger] hadware
wallets with Oasis Network. Ledger Live software doesn't support Oasis (ROSE)
tokens natively yet. In this guide we will install Oasis app via Ledger Live to
open and access wallet with one or multiple accounts via our official
[Oasis Wallet - Web].

## Setup your Ledger device and Install Oasis App

1. To use your [Ledger] wallet to hold your ROSE tokens, you will have to
install [Oasis app] on your Ledger wallet via [Ledger Live]'s Manager. You need
to connect your Ledger to your device and unlock it with your PIN code first.

![Unlock ledger](<../../images/wallet/ledger/1_unlock_ledger.png>)

> 2. Currently, you will have to enable _Developer mode_ by going to _Settings ->

Reviewer 2: ```suggestion
2. Currently, you will have to enable _Developer mode_ by going to _Settings →
```

> Experimental features_ -> enable _Developer mode_ toggle, to be able to find

Reviewer 2: ```suggestion
Experimental features_ → enable _Developer mode_ toggle, to be able to find
```

while your device may lock for safety reasons and you will need to unlock it to
perform this and subsequent steps.

:::

3. After connecting your Ledger to Oasis Wallet - Web another pop-up will appear
where you can choose to open one or more `oasis1` accounts derived from the seed
stored on your Ledger.

![Oasis Wallet - Web -> Select one or more accounts](<../../images/wallet/ledger/9_oasis_wallet_web_select_accounts_to_open.png>)

4. The account from your Ledger device is now opened. If you import multiple
accounts, you can switch between them by clicking on the account address in
the top-right corner.
Our demo account is empty.

![Oasis Wallet - Web -> Selected accounts are opened](<../../images/wallet/ledger/10_oasis_wallet_web_opene_ledger_account.png>)

5. Now you can use your Ledger to receive, send or delegate ROSE.
In this example, we have received 111 ROSE to our Ledger account.

![Oasis Wallet - Web -> Exploring transactions](<../../images/wallet/ledger/11_oasis_wallet_web_received_rose_on_ledger_account.png>)

### Using Oasis CLI Tools

This is the most powerful option that allows performing not just token-related

> tasks(sending, staking, ParaTime deposits and withdrawals), but also generating

Reviewer 2: ```suggestion
tasks (sending, staking, ParaTime deposits and withdrawals), but also generating
```

:::

3. To find the Oasis app in the Ledger Live App catalog, you need to Allow Ledger Manager on your Ledger device first, then you will be able to click App catalog and search for _Oasis_:

> ![](<../../images/wallet/ledger/3_allow_ledger_manager.png>)

Reviewer 2: And here 3x

![](<../../images/wallet/ledger/4_search_apps.png>)
![](<../../images/wallet/ledger/6_search_results_oasis_install.png>)

:::caution

> Make sure you install at least version 2.3.2 of the Oasis app.

Reviewer 2: Below please add the instructions how to launch the Oasis app on Ledger and add a photo of "Oasis Ready" state.

:::

> 

Reviewer 2: Remove double blank line (sorry, no markdown linter set up yet ;) ).

Once installed, the Oasis app will use the [BIP 39](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki) mnemonic seed stored on your Ledger wallet to generate the private & public key pairs for your accounts.

## Manage Your Tokens

### Using Oasis Wallets

> This is a simpler option since it allows you to connect to your Ledger wallet via our official wallets. In our example we will use Oasis Wallet - Web. For how to use our official wallets please use instructions in the [Oasis Wallets](../oasis-wallets/README.mdx) doc.

Reviewer 3: "This is a simpler option since it allows you to connect to your Ledger wallet via our official wallets." doesn't really explain why it is simpler, like it did before.

Author: "This is a simpler option since it allows you to connect to your Ledger wallet via web application or browser extension." applied.

Author: "This is a simpler option since it allows you to connect to your Ledger wallet via web application or browser extension." applied.

> This is a simpler option since it allows you to connect to your Ledger wallet via web application or browser extension. In our example we will use Oasis Wallet - Web. For how to use our official wallets please use instructions in the [Oasis Wallets](../oasis-wallets/README.mdx) doc.

Reviewer 2: ```suggestion
This is a simpler option since it allows you to connect to your Ledger wallet via a web application or a browser extension.

In the example below we will use Oasis Wallet - Web. To learn more about other features of the Oasis Wallets, please read the [Oasis Wallets](../oasis-wallets/README.mdx) doc.
```

> 1. Navigate to [wallet.oasisprotocol.org](https://wallet.oasisprotocol.org) and click _Open wallet_. Then click _Ledger_ when asked How do you want to open your wallet.

Reviewer 2: ```suggestion
1. Navigate to [wallet.oasisprotocol.org](https://wallet.oasisprotocol.org) and click on the _Open wallet_ button. Then, click on _Ledger_ when asked how to open your wallet.
```

> ![](<../../images/wallet/ledger/7_oasis_wallet_web_open_ledger.png>)

Reviewer 2: Add image description.

> 2. In next step when clicking _Select the wallets to open_ button a pop-up window in your browser will appear where you will have to select your Ledger device and then click _Connect_.

Reviewer 2: ```suggestion
2. In the next step, click on the _Select the wallets to open_ button. Your browser will open a pop-up window where you will have to select your Ledger device. Finally, click on _Connect_.
```

> ![](<../../images/wallet/ledger/8b_oasis_wallet_web_select_ledger_device_connect.png>)

Reviewer 2: Add image description.

:::caution

> Make sure you open Oasis app on your Ledger before clicking _Select the wallets to open_ button. To do that navigate to Oasis app on Ledger device and use two buttons to open Oasis app, then confirm Pending Ledger review again with two buttons and Oasis Ready text should appear.

Reviewer 2: ```suggestion
Make sure you open the Oasis app on your Ledger before clicking on the _Select the wallets to open_ button. To do so, navigate to the Oasis app in the application list of your Ledger device and press the two buttons to open it. Then, confirm the _Pending Ledger review_ warning in the same way and the _Oasis Ready_ message should appear.
```

Also, this should be moved as a normal paragraph at the end of the previous chapter (see my comment above).

Here, I would just add a caution like this

```
This step requires that your Ledger shows the _Oasis Ready_ message. After a while your device may lock for safety reasons and you will need to unlock it to perform this and subsequent steps.
```

> Make sure you open Oasis app on your Ledger before clicking _Select the wallets to open_ button. To do that navigate to Oasis app on Ledger device and use two buttons to open Oasis app, then confirm Pending Ledger review again with two buttons and Oasis Ready text should appear.

Reviewer: ```suggestion
Make sure you open Oasis app on your Ledger before clicking _Select the wallets to open_ button. To do that navigate to Oasis app on Ledger device and use two buttons to open Oasis app, then confirm _Pending Ledger review_ again with two buttons and Oasis Ready text should appear.
```

:::

> 3. After Connecting your Ledger to Oasis Wallet - Web a new pop-up will appear where you will choose one or multiple oasis1 wallets to be opened.

Reviewer 2: ```suggestion
3. After connecting your Ledger to Oasis Wallet - Web another pop-up will appear where you can choose to open one or more `oasis1` wallets derived from the seed stored on your Ledger.
```

> 3. After Connecting your Ledger to Oasis Wallet - Web a new pop-up will appear where you will choose one or multiple oasis1 wallets to be opened.

Reviewer: I would prefer to be very precise w.r.t to wallets vs. account. Here, one or more accounts (from the same wallet) can be opened.

The Oasis Wallet - Web's UI needs to be updated to mention accounts as well.

Reviewer 3: for reference: @matevz added https://github.com/oasisprotocol/oasis-wallet-web/pull/814

> ![](<../../images/wallet/ledger/9_oasis_wallet_web_select_wallets_to_open.png>)

Reviewer 2: Add image description.

> 5. You oasis1 wallet from Ledger device is opened and you can switch between multiple wallets by clicking your wallet address near Mainnet/Testnet swith button. Our demo wallet is empty.

Reviewer 3: Typo "You oasis1 wallet" and "swith"

Author: "Your" and "switch" applied.

Author: "Your" and "switch" applied.

> 5. Your oasis1 wallet from Ledger device is opened and you can switch beetween multiple wallets by clicking your wallet address near Mainnet/Testnet switch button. Our demo wallet is empty.

Reviewer 2: ```suggestion
5. Your oasis1 wallet from Ledger device is opened and you can switch between multiple wallets by clicking on your wallet address in the top-right corner. Our demo wallet is empty.
```

> ![](<../../images/wallet/ledger/10_oasis_wallet_web_opene_ledger_account.png>)

Reviewer 3: Typo on screenshot: "beetween"

Author: "beetween" applied.

Author: "beetween" applied.

> ![](<../../images/wallet/ledger/10_oasis_wallet_web_opene_ledger_account.png>)

Reviewer 2: Add image description.

> 6. We have sent 111 ROSE to our Ledger oasis1 wallet and ROSE appear in our Oasis Wallet - Web imediatelly after sent.

Reviewer 3: I don't like the sentence :D

Author: Rephrased. :))

Author: Rephrased. :))

> ![](<../../images/wallet/ledger/11_oasis_wallet_web_received_rose_on_ledger_account.png>)

Reviewer 2: Add image description.
